### PR TITLE
[automation/python] - Fix stack settings serialization

### PR DESCRIPTION
**Pull request by komalali**
Marking this no-changelog-required because we're in a weird in-between state and this will go out as a patch fix so the changelog will be added during release.

Changelog entry:

Bugfix:

- [automation/python] - Fix serialization bug in `StackSettings`

### CODE REVIEW

**Comment by stack72** [comment]
This is going into master and then being cherry picked to a patch of v2.25.0 so we don’t need the changelog specifically in master right now

**Review by stack72 — APPROVED** [review]

File: sdk/python/lib/pulumi/automation/_stack_settings.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class StackSettings:
     secrets_provider: Optional[str]
     encrypted_key: Optional[str]
     encryption_salt: Optional[str]
-    config: Optional[Mapping[str, Any]]
+    config: Optional[Dict[str, Union[str, StackSettingsSecureConfigValue]]]
```
Comment (leezen) [inline sdk/python/lib/pulumi/automation/_stack_settings.py:31]: Shouldn't this technically be `Any` though since we can have nested objects in the config?

Comment (komalali) [reply sdk/python/lib/pulumi/automation/_stack_settings.py:31]: For now, in automation API, we don't unmarshal objects and arrays from config, and they are read in as strings. There's an [open issue](https://github.com/pulumi/pulumi/issues/5506) tracking this. So, this matches the current reality and the other SDKs.

Comment (leezen) [reply sdk/python/lib/pulumi/automation/_stack_settings.py:31]: Ah, right. Thanks for clarifying.

**Review by leezen — APPROVED** [review]

File: sdk/python/lib/pulumi/automation/_stack_settings.py (hunk 1 above)
Comment (komalali) [reply sdk/python/lib/pulumi/automation/_stack_settings.py:31]: Actually, you're right, kinda. I was thinking of `pulumi config` commands in my earlier answer. 

When the settings are parsed from json/yaml, in python the nested objects will parse to dicts and lists. So we should be recursing through if it's an object or a list to find the "secure" and regular config values. That means it's not just the type that's wrong, but also the serde for nested stack settings config in general 😞 

Gonna have to look at this with fresh eyes in the morning

Comment (komalali) [reply sdk/python/lib/pulumi/automation/_stack_settings.py]: Okay, I updated the types. The deserialization is still not perfect but the issue falls under the umbrella covered in #5506.

**Pull request merged**